provider/cassandra.go

```diff
@@ -0,0 +1,206 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
```

Comment:
If you split files again, should make different PRs for the new code and file refactor. Same goes for @RiddhiBagadiaa , but this is fine for now